waspls/src/Wasp/LSP/ServerState.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Wasp.LSP.ServerState
+  ( ServerState (..),
+    sourceString,
+    diagnostics,
+    cst,
+  )
+where
+
+import Control.Lens (makeClassy)
+import Data.Default (Default (def))
+import qualified Language.LSP.Types as LSP
+import Wasp.Backend.ConcreteSyntax (SyntaxNode)
+
+-- | LSP State preserved between handlers
+--
+-- Recommended to use the lenses for accessing the fields.
+data ServerState = ServerState
+  { -- | Source text for wasp file
+    _sourceString :: String,
```

Comment:
Which one? Doesn't server deal with a lot of files?

Reply:
The server currently assumes only one wasp file exists in a project, which is true (for now) in wasp projects. I've added this assumption to the description here.

Reply:
How will this change once we have multiple files -> what will we want to store then, here in state?
If we are going to be storing only the state of the latest analyzed file, then we could rename this right now to smth like `currentWaspSource` or `lastAnalyzedWaspSource` or smth like that.
If it is going to need to be more complex, like a map of sources and their data, then probably best not to complicate for now.

Reply:
When we start having multiple files, this will switch to something more complicated (at least a map of filepath to data). In the meantime, I've switched this to be named `currentWaspSource`.